dags/elsevier/trigger_file_processing.py

```diff
@@ -0,0 +1,62 @@
+import base64
+import io
+import os
+
+from airflow.api.common import trigger_dag
+from common.pull_ftp import _generate_id
+from common.repository import IRepository
+from common.utils import parse_without_names_spaces, process_archive
+from elsevier.metadata_parser import ElsevierMetadataParser
+from structlog import PrintLogger
+
+
+def trigger_file_processing_elsevier(
+    publisher: str,
+    repo: IRepository,
+    logger: PrintLogger,
+    filenames=None,
+):
+    files = []
+    for filename in filenames:
+        logger.msg("Running processing.", filename=filename)
+        file_bytes = repo.get_by_id(filename)
+        (archive_file_content, s3_filename) = next(
+            process_archive(
+                file_bytes=file_bytes,
+                file_name=filename,
+                only_specific_file="dataset.xml",
+            )
+        )
+        dataset_file = parse_without_names_spaces(archive_file_content.decode("utf-8"))
+        parser = ElsevierMetadataParser(file_path=filename)
+        parsed_articles = parser.parse(dataset_file)
+        for parsed_article in parsed_articles:
+            (file_content, full_file_path) = next(
+                process_archive(
+                    file_bytes=file_bytes,
+                    file_name=s3_filename,
+                    only_specific_file=parsed_article["files"]["xml"],
+                    form_s3_key=False,
+                )
+            )
```

Comment:
We already have the the file_name right? We can simplify all this by just ignoring the `full_file_path` which is actually the `filename` for this case according to the code of `process_zip_file`
```suggestion
            file_content, _ = next(
                process_archive(
                    file_bytes=file_bytes,
                    file_name=s3_filename,
                    only_specific_file=parsed_article["files"]["xml"],
                    form_s3_key=False,
                )
            )
```

Reply:
No, we don't have full_file_path. filename is the name of zip file, and full_file_path is the path of file, which is inside zip/tar